src/processors/models/processorsManager.js

```diff
@@ -34,12 +34,39 @@ module.exports.getProcessor = async function(processorId) {
     if (processor) {
         return processor;
     } else {
-        const error = new Error(ERROR_MESSAGES.NOT_FOUND);
-        error.statusCode = 404;
+        const error = generateProcessorNotFoundError();
         throw error;
     }
 };
 
 module.exports.deleteProcessor = async function (processorId) {
     return databaseConnector.deleteProcessor(processorId);
 };
+
+module.exports.updateDownloadJSProcessor = async function(processorId) {
+    const processor = await databaseConnector.getProcessor(processorId);
+    try {
+        if (!processor) {
+            const error = generateProcessorNotFoundError();
+            throw error;
+        }
+        if (!processor.file_url) {
+            throw new Error('option not available for static js content processor');
+        }
+        let newJavascriptProcessorContent = await fileManager.downloadFile(processor.file_url);
+        fileManager.validateJavascriptContent(newJavascriptProcessorContent);
+        processor.javascript = newJavascriptProcessorContent;
+        await databaseConnector.insertProcessor(processorId, processor);
```

Comment:
Want to update the current processor with the new content and not insert new one

Reply:
yeah, just building the template for the PR